src/Voron/Platform/Posix/PosixTempMemoryMapPager.cs

```diff
@@ -126,11 +128,15 @@ private PagerState CreatePagerState()
                 Syscall.ThrowLastError(err, "mmap on " + FileName);
             }
             NativeMemory.RegisterFileMapping(FileName.FullPath, startingBaseAddressPtr, _totalAllocationSize);
+
+            bool isResidentInRam = Memory.PosixCheckForResidencyInRam(startingBaseAddressPtr, _totalAllocationSize);
+
             var allocationInfo = new PagerState.AllocationInfo
             {
                 BaseAddress = (byte*)startingBaseAddressPtr.ToPointer(),
                 Size = _totalAllocationSize,
-                MappedFile = null
+                MappedFile = null,
+                IsNotResidentInRam = !isResidentInRam
```

Comment:
Why do we need this? This is almost always going to be true, because we prefetch on startup